test/battle/trainer_slides.c

```diff
@@ -0,0 +1,210 @@
+        PLAYER(SPECIES_BULBASAUR);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_VINE_WHIP); }
+    } SCENE {
+        MESSAGE("Bulbasaur used Vine Whip!");
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_VINE_WHIP, player);
+        MESSAGE("Player lands their first STAB move.{PAUSE_UNTIL_PRESS}");
+    }
+}
+
+SINGLE_BATTLE_TEST("Trainer Slide: Player Lands First Super Effective Hit")
+{
+    gBattleTestRunnerState->data.recordedBattle.opponentA = TRAINER_SLIDE_PLAYER_LANDS_FIRST_SUPER_EFFECTIVE_HIT;
+
+    GIVEN {
+        ASSUME(GetMoveType(MOVE_BITE) == TYPE_DARK);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_BITE); }
+    } SCENE {
+        MESSAGE("It's super effective!");
+        MESSAGE("This message plays after the player lands their first super effective hit.{PAUSE_UNTIL_PRESS}");
+    }
+}
+
+SINGLE_BATTLE_TEST("Trainer Slide: Player Lands First Down")
+{
+    gBattleTestRunnerState->data.recordedBattle.opponentA = TRAINER_SLIDE_PLAYER_LANDS_FIRST_DOWN;
+
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WYNAUT);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_HEALING_WISH); SEND_OUT(opponent,1); }
+    } SCENE {
+        MESSAGE("The opposing Wobbuffet fainted!");
+        MESSAGE("This message plays after the player KOs one enemy mon.{PAUSE_UNTIL_PRESS}");
+    }
+}
+
+SINGLE_BATTLE_TEST("Trainer Slide: Enemy Mon Unaffected")
+{
+    gBattleTestRunnerState->data.recordedBattle.opponentA = TRAINER_SLIDE_ENEMY_MON_UNAFFECTED;
+    GIVEN {
+        ASSUME(B_SHEER_COLD_IMMUNITY >= GEN_7);
+        ASSUME(gSpeciesInfo[SPECIES_GLALIE].types[0] == TYPE_ICE);
+        PLAYER(SPECIES_WYNAUT);
+        OPPONENT(SPECIES_GLALIE);
+    } WHEN {
+        TURN { MOVE(player, MOVE_SHEER_COLD); }
+    } SCENE {
+        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_SHEER_COLD, player);
+        MESSAGE("It doesn't affect the opposing Glalie…");
+        MESSAGE("Player attacked enemy with ineffective move.{PAUSE_UNTIL_PRESS}");
+    }
+}
+
+SINGLE_BATTLE_TEST("Trainer Slide: Last Switchin")
+{
+    gBattleTestRunnerState->data.recordedBattle.opponentA = TRAINER_SLIDE_LAST_SWITCHIN;
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WYNAUT);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_HEALING_WISH); SEND_OUT(opponent,1); }
+    } SCENE {
+        MESSAGE("The opposing Wobbuffet fainted!");
+        MESSAGE("This message plays after the enemy switches in their last Pokemon.{PAUSE_UNTIL_PRESS}");
+    }
+}
+
+SINGLE_BATTLE_TEST("Trainer Slide: Last Half Hp")
+{
+    gBattleTestRunnerState->data.recordedBattle.opponentA = TRAINER_SLIDE_LAST_HALF_HP;
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Attack(515);}
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_LIQUIDATION); }
```

Comment:
```suggestion
        ASSUME(gMovesInfo[MOVE_SUPER_FANG].effect == EFFECT_SUPER_FANG);
        ASSUME(gSpeciesInfo[SPECIES_WOBBUFFET].baseHP == 190);
        PLAYER(SPECIES_WOBBUFFET);
        OPPONENT(SPECIES_WOBBUFFET);
    } WHEN {
        TURN { MOVE(player, MOVE_SUPER_FANG); }
```
Using Super Fang is more robust to check if the 50% HP threshold it met, Wobbuffet has an even HP at level 100 by default, so checking that the `baseHP` is unchanged should be enough.